cpp/src/column/column_device_view.cu

```diff
@@ -44,12 +45,10 @@ std::unique_ptr<ColumnDeviceView, std::function<void(ColumnDeviceView*)>>
 create_device_view_from_view(ColumnView const& source, rmm::cuda_stream_view stream)
 {
   size_type num_children = source.num_children();
-  // First calculate the size of memory needed to hold the
-  // child columns. This is done by calling extent()
-  // for each of the children.
-  auto get_extent = thrust::make_transform_iterator(
-    thrust::make_counting_iterator(0),
-    [&source](auto i) { return ColumnDeviceView::extent(source.child(i)); });
+  // First calculate the size of memory needed to hold the child columns. This is done by calling
+  // extent() for each of the children.
+  auto get_extent = cudf::detail::make_counting_transform_iterator(
+    0, [&source](auto i) { return ColumnDeviceView::extent(source.child(i)); });
```

Comment:
Similar to other changes, I think this would be more natural. 
```suggestion
  auto get_extent = thrust::make_transform_iterator(source.child_begin(), [](auto const& child) { return ColumnDeviceView::extent(child); });
```

Reply:
hmm, it is upset about `~mutable_column_view` 
```
device/per_device_resource.hpp (89): error: calling a __host__ function("std::vector< ::cudf::mutable_column_view, ::std::allocator< ::cudf::mutable_column_view> > ::~vector") from a __host__ __device__ function("cudf::mutable_column_view::~mutable_column_view") is not allowed
```

Reply:
Ah, one of those. Those are really annoying and hard to diagnose sometimes. I guess just don't bother with it. I am a little surprised it's coming up in this situation. Is the argument to the lambda being passed as a reference or value?

Reply:
I originally passed as `const&`. Just tried by value with same result.